### PR TITLE
test: 81% coverage on blockstore

### PR DESCRIPTION
**Pull request by Kubuxu**
Coverage report available at: https://ipfs.io/ipfs/QmTuMtwGCfHrbYyZdQ1RaGNwS2MGsmAkjA8AaB69N7Ya1g/coverage.html#file0

Part of #3053

License: MIT
Signed-off-by: Jakub Sztandera <kubuxu@protonmail.ch>

### CODE REVIEW

**Comment by Kubuxu** [comment]
Docker tests failed but it isn't connected.

File: blocks/blockstore/blockstore_test.go [hunk 1]
```diff
@@ -57,16 +57,21 @@ func TestRuntimeHashing(t *testing.T) {
 	bs := NewBlockstore(ds_sync.MutexWrap(ds.NewMapDatastore()))
 	bl := blocks.NewBlock([]byte("some data"))
 	blBad, err := blocks.NewBlockWithHash([]byte("some other data"), bl.Key().ToMultihash())
+	bl2 := blocks.NewBlock([]byte("some other data"))
```
Comment (whyrusleeping) [inline blocks/blockstore/blockstore_test.go]: why is this here before the error check from the previous statement?

File: blocks/blockstore/caching_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package blockstore
+
+import "testing"
+
+func TestCachingOptsLessThanZero(t *testing.T) {
+	opts := DefaultCacheOpts()
+	opts.HasARCCacheSize = -1
+
+	if _, err := CachedBlockstore(nil, nil, opts); err == nil {
+		t.Fatal()
```
Comment (whyrusleeping) [inline blocks/blockstore/caching_test.go]: these fatals should ideally give us some small idea of why it failed

File: blocks/blockstore/blockstore_test.go (hunk 1 above)
Comment (Kubuxu) [inline blocks/blockstore/blockstore_test.go]: because it should always success and it is checked in other tests, will add check either way.

Comment (whyrusleeping) [inline blocks/blockstore/blockstore_test.go]: still, you want to check errors right as they are created, otherwise it just looks sloppy.

**Comment by Kubuxu** [comment]
Updated.

File: blocks/blockstore/caching_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package blockstore
+
+import "testing"
+
+func TestCachingOptsLessThanZero(t *testing.T) {
+	opts := DefaultCacheOpts()
+	opts.HasARCCacheSize = -1
+
+	if _, err := CachedBlockstore(nil, nil, opts); err == nil {
+		t.Error("wrong ARC setting was not detected")
```
Comment (whyrusleeping) [inline blocks/blockstore/caching_test.go]: Any reason for `Error` over `Fatal`?

Comment (Kubuxu) [inline blocks/blockstore/caching_test.go]: Those test cases are not dependent on each other, so using `Error` all test cases will checked and printed.

`Error = Print + Fail`
`Fatal = Print + FailNow`

Comment (whyrusleeping) [inline blocks/blockstore/caching_test.go]: I guess thats fair.

File: blocks/blockstore/arc_cache_test.go [hunk 4]
```diff
@@ -42,26 +59,119 @@ func TestRemoveCacheEntryOnDelete(t *testing.T) {
...
+	}
+
+	trap("has hit datastore", cd, t)
+
+	if has, err := arc.Has(exampleBlock.Key()); !has || err != nil {
+		t.Fatal("has returned invalid result")
+	}
+}
+
+func TestGetAndDeleteFalseShortCirciot(t *testing.T) {
```
Comment (whyrusleeping) [inline blocks/blockstore/arc_cache_test.go]: typo here, sorry.

**Comment by whyrusleeping** [comment]
just a typo then LGTM

**Comment by Kubuxu** [comment]
Done.
